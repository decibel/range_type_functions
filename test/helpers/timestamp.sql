INSERT INTO test VALUES
    ( 'now() +/- 1us'
      , array[
          now()
          , now() + interval '1 us'
          , now() - interval '1 us'
        ]
      , :range_type(
          now():::subtype - interval '1 us'
          , now():::subtype + interval '1 us'
          , '[]'
      )
      , :range_type(
          now():::subtype - interval '1 us'
          , now():::subtype + interval '1 us'
          , '[)'
      )
    )

  , ( 'timezone test'
      , array[
          '2000-1-1'::timestamptz AT TIME ZONE 'CST6CDT'
          , '2000-1-1'::timestamptz AT TIME ZONE 'MST7MDT'
          , '2000-1-1'::timestamptz AT TIME ZONE 'EST5EDT'
        ]
      , :range_type(
        '2000-1-1'::timestamptz AT TIME ZONE 'MST7MDT'
        , '2000-1-1'::timestamptz AT TIME ZONE 'EST5EDT'
        , '[]'
      )
      , :range_type(
        '2000-1-1'::timestamptz AT TIME ZONE 'MST7MDT'
        , '2000-1-1'::timestamptz AT TIME ZONE 'EST5EDT'
        , '[)'
      )
    )
;

-- vi: expandtab ts=2 sw=2
