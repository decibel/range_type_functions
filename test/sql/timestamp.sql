\set ECHO none

\set range_type tsrange
\set subtype timestamp
\i test/helpers/type_setup.sql

\i test/helpers/timestamp.sql

\i test/helpers/type_run.sql

-- vi: expandtab ts=2 sw=2
