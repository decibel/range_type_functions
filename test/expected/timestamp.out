\set ECHO none
1..4
ok 1 - to_range from now() +/- 1us
ok 2 - to_range from timezone test
ok 3 - now() +/- 1us
ok 4 - timezone test
# TRANSACTION INTENTIONALLY LEFT OPEN!
